internal/controller/status/status.go

```diff
@@ -33,35 +30,41 @@ func SyncNfsPvcStatus(ctx context.Context, nfspvc danaiov1alpha1.NfsPvc, log log
 	pvPhase := getPvStatus(ctx, nfspvc, k8sClient)
 
 	if pvcPhase != nfspvc.Status.PvcPhase || pvPhase != nfspvc.Status.PvPhase {
-		// Use retry on conflict to update the nfspvc status.
-		err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-			nfspvcObject.Status.PvcPhase = pvcPhase
-			nfspvcObject.Status.PvPhase = pvPhase
+		return UpdatePvcStatus(ctx, pvcPhase, pvPhase, nfspvcObject, k8sClient)
+	}
+	return nil
+}
+
+// UpdatePvcStatus Update the status of the pvc.
+func UpdatePvcStatus(ctx context.Context, pvcPhase string, pvPhase string, nfspvcObject danaiov1alpha1.NfsPvc, k8sClient client.Client) error {
+	// Use retry on conflict to update the nfspvc status.
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		nfspvcObject.Status.PvcPhase = pvcPhase
+		nfspvcObject.Status.PvPhase = pvPhase
 
-			updateErr := k8sClient.Status().Update(ctx, &nfspvcObject)
-			if errors.IsConflict(updateErr) {
-				// Conflict occurred, let's re-fetch the latest version of NFSPVC and retry the update.
-				if getErr := k8sClient.Get(ctx, types.NamespacedName{Name: nfspvcObject.Name, Namespace: nfspvcObject.Namespace}, &nfspvcObject); getErr != nil {
-					return getErr
-				}
+		updateErr := k8sClient.Status().Update(ctx, &nfspvcObject)
+		if errors.IsConflict(updateErr) {
+			// Conflict occurred, let's re-fetch the latest version of NFSPVC and retry the update.
+			if getErr := k8sClient.Get(ctx, types.NamespacedName{Name: nfspvcObject.Name, Namespace: nfspvcObject.Namespace}, &nfspvcObject); getErr != nil {
+				return getErr
 			}
-			return updateErr
-		})
-		return err
-	}
+		}
+		return updateErr
+	})
+	//Should be nil
```

Comment:
I meant for this comment to be explanatory. If everything went ok, err = nil but since we return err, the reader might get confused. I will replace this with more indicative logic